nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -412,17 +412,20 @@ void CollisionMonitor::process(const Velocity & cmd_vel_in, const std_msgs::msg:
   }
 
   // Points array collected from different data sources in a robot base frame
-  std::vector<Point> collision_points;
+  std::unordered_map<std::string, std::vector<Point>> sources_collision_points_map;
 
   // By default - there is no action
   Action robot_action{DO_NOTHING, cmd_vel_in, ""};
   // Polygon causing robot action (if any)
   std::shared_ptr<Polygon> action_polygon;
 
-  // Fill collision_points array from different data sources
+  // Fill collision points array from different data sources
   for (std::shared_ptr<Source> source : sources_) {
+    std::vector<Point> collision_points;
+    auto iter = sources_collision_points_map.insert({source->getSourceName(), collision_points});
```

Comment:
```suggestion
    auto iter = sources_collision_points_map.insert({source->getSourceName(), std::vector<Point>()});
```

I think this would work, no? Removes intermediate variable that wouldn't need to be stored in scope then

Reply:
You're right, it's updated